modules/javafx.controls/src/test/java/test/javafx/scene/control/ControlPropertiesTest.java

```diff
@@ -0,0 +1,219 @@
+            //HTMLEditor.class,
+            Hyperlink.class,
+            IndexedCell.class,
+            InputField.class,
+            IntegerField.class,
+            Labeled.class,
+            Label.class,
+            ListCell.class,
+            ListView.class,
+            MenuBar.class,
+            MenuButton.class,
+            Pagination.class,
+            PasswordField.class,
+            ProgressIndicator.class,
+            ProgressBarTreeTableCell.class,
+            RadioButton.class,
+            ScrollBar.class,
+            ScrollPane.class,
+            Separator.class,
+            SeparatorMenuItem.class,
+            Slider.class,
+            Spinner.class,
+            SplitPane.class,
+            TableRow.class,
+            TableView.class,
+            TabPane.class,
+            TextArea.class,
+            TextField.class,
+            TextFieldTreeCell.class,
+            TextFieldTreeTableCell.class,
+            TitledPane.class,
+            ToggleButton.class,
+            TreeCell.class,
+            TreeTableCell.class,
+            TreeTableRow.class,
+            TreeTableView.class,
+            WebColorField.class
+        );
+    }
+
+    /**
+     * Tests for missing final keyword in Control properties and their accessors/mutators.
+     */
+    @Test
+    public void testMissingFinalMethods() {
+        for (Class c : allControlClasses()) {
+            check(c);
+        }
+    }
+
+    private void check(Class cls) {
+        Method[] publicMethods = cls.getMethods();
+        for (Method m : publicMethods) {
+            String name = m.getName();
+            if (name.endsWith("Property") && (m.getParameterCount() == 0)) {
+                checkModifiers(m);
+
+                String propName = name.substring(0, name.length() - "Property".length());
+                check(publicMethods, propName, "get", 0);
+                check(publicMethods, propName, "set", 1);
+                check(publicMethods, propName, "is", 0);
+            }
+        }
+    }
+
+    private void check(Method[] methods, String propName, String prefix, int numArgs) {
+        StringBuilder sb = new StringBuilder(64);
+        sb.append(prefix);
+        sb.append(Character.toUpperCase(propName.charAt(0)));
+        sb.append(propName, 1, propName.length());
+
+        String name = sb.toString();
+        for (Method m : methods) {
+            if (m.getParameterCount() == numArgs) {
+                if (name.equals(m.getName())) {
+                    checkModifiers(m);
+                    return;
+                }
+            }
+        }
+    }
+
+    private void checkModifiers(Method m) {
+        int mod = m.getModifiers();
+        if (Modifier.isPublic(mod) && !Modifier.isFinal(mod)) {
+            String msg = m + " is not final.";
+            if (FAIL_FAST) {
+                throw new AssertionError(msg);
+            } else {
+                System.err.println(msg);
+            }
```

Comment:
No need for an `else` clause since `throw` will exit the scope of the method if it's reached.

Reply:
personal style.